src/Akka.Streams.Kafka.Tests/Integration/CommittableSourceIntegrationTests.cs

```diff
@@ -0,0 +1,160 @@
+                .WithBootstrapServers(KafkaUrl)
+                .WithProperty("auto.offset.reset", "earliest")
+                .WithGroupId(group);
+        }
+
+        [Fact]
+        public async Task CommitableSource_consumes_messages_from_Producer_without_commits()
+        {
+            int elementsCount = 100;
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+
+            await GivenInitializedTopic(topic1);
+
+            await Source
+                .From(Enumerable.Range(1, elementsCount))
+                .Select(elem => new ProduceRecord<Null, string>(topic1, null, elem.ToString()))
+                .RunWith(Dsl.Producer.PlainSink(ProducerSettings), _materializer);
+
+            var consumerSettings = CreateConsumerSettings(group1);
+
+            var probe = Dsl.Consumer
+                .CommitableSource(consumerSettings, Subscriptions.Assignment(new TopicPartition(topic1, 0)))
+                .Where(c => !c.Record.Value.Equals(InitialMsg))
+                .Select(c => c.Record.Value)
+                .RunWith(this.SinkProbe<string>(), _materializer);
+
+            probe.Request(elementsCount);
+            foreach (var i in Enumerable.Range(1, elementsCount).Select(c => c.ToString()))
+                probe.ExpectNext(i, TimeSpan.FromSeconds(10));
+
+            probe.Cancel();
+        }
+
+        [Fact]
+        public async Task CommitableSource_resume_from_commited_offset()
+        {
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+            var group2 = CreateGroup(2);
+
+            await GivenInitializedTopic(topic1);
+
+            await Source
+                .From(Enumerable.Range(1, 100))
+                .Select(elem => new ProduceRecord<Null, string>(topic1, null, elem.ToString()))
+                .RunWith(Dsl.Producer.PlainSink(ProducerSettings), _materializer);
+
+            var consumerSettings = CreateConsumerSettings(group1);
+            var committedElements = new ConcurrentQueue<string>();
+
+            var (_, probe1) = Dsl.Consumer.CommitableSource(consumerSettings, Subscriptions.Assignment(new TopicPartition(topic1, 0)))
+                .WhereNot(c => c.Record.Value == InitialMsg)
+                .SelectAsync(10, elem =>
+                {
+                    return elem.CommitableOffset.Commit().ContinueWith(t =>
+                    {
+                        committedElements.Enqueue(elem.Record.Value);
+                        return Done.Instance;
+                    });
+                })
+                .ToMaterialized(this.SinkProbe<Done>(), Keep.Both)
+                .Run(_materializer);
+
+            probe1.Request(25);
+
+            foreach (var _ in Enumerable.Range(1, 25))
+            {
+                probe1.ExpectNext(Done.Instance, TimeSpan.FromSeconds(10));
+            }
+                
+            probe1.Cancel();
+
+            // Await.result(control.isShutdown, remainingOrDefault)
```

Comment:
Control is not implemented at the moment

Reply:
My idea was to make materialized value implement a `IDisposable` object - this way it can be set inside using block, so we don't have to worry about not-disposed connection in case of errors.

Reply:
Or we could use `CancellationTokenSource` here